app/assets/javascripts/discourse/tests/unit/services/presence-test.js

```diff
@@ -242,51 +243,50 @@ acceptance("Presence - Subscribing", function (needs) {
   });
 });
 
-acceptance("Presence - Entering and Leaving", function (needs) {
-  needs.user();
+module("Unit | Service | presence | entering and leaving", function (hooks) {
+  setupTest(hooks);
 
   let responseStartPromise;
   let resolveResponseStartPromise;
   let responseWaitPromise;
 
   const requests = [];
 
-  needs.hooks.beforeEach(() => {
+  hooks.beforeEach(function () {
     responseStartPromise = new Promise(
       (resolve) => (resolveResponseStartPromise = resolve)
     );
-  });
-
-  needs.hooks.afterEach(() => {
-    requests.clear();
-    responseWaitPromise = null;
-    responseStartPromise = null;
-  });
 
-  needs.pretender((server, helper) => {
-    server.post("/presence/update", async (request) => {
+    pretender.post("/presence/update", async (request) => {
       resolveResponseStartPromise();
       await responseWaitPromise;
 
       const body = new URLSearchParams(request.requestBody);
       requests.push(body);
 
-      const response = {};
+      const result = {};
       const channelsRequested = body.getAll("present_channels[]");
       channelsRequested.forEach((c) => {
         if (c.startsWith("/test/")) {
-          response[c] = true;
+          result[c] = true;
         } else {
-          response[c] = false;
+          result[c] = false;
         }
       });
 
-      return helper.response(response);
+      return response(result);
     });
   });
 
+  hooks.afterEach(function () {
+    requests.clear();
+    responseWaitPromise = null;
+    responseStartPromise = null;
+  });
+
   test("can join and leave channels", async function (assert) {
-    const presenceService = this.container.lookup("service:presence");
+    const presenceService = getOwner(this).lookup("service:presence");
+    presenceService.currentUser = currentUser();
```

Comment:
This seems a little strange - no `service:current-user` injected automatically here?

Reply:
pre-initializers don't run in unit tests, so it's not injected. we already use the pattern of manually setting currentUser prop in e.g. `document-title` service tests.